xgi/algorithms/properties.py

```diff
@@ -33,19 +33,22 @@ def equal(H1, H2, compare_ids=True, compare_attrs=True):
         edges2_with_counts = defaultdict(lambda: 0)
         for e in H2.edges.members():
             edges2_with_counts[frozenset(e)] += 1
-
+        print(dict(edges1_with_counts), dict(edges2_with_counts))
         if edges1_with_counts != edges2_with_counts:
+            print("hi")
             return False
 
     if compare_attrs:
-        if H1._hypergraph != H2._hypergraph:
+        if H1._net_attr != H2._net_attr:
             return False
 
-        if H1._node_attrs != H2._node_attrs:
+        if H1._node_attr != H2._node_attr:
             return False
 
-        if H1._edge_attrs != H2._edge_attrs:
-            return False
+        # there's no sense in comparing edge attrs if we're not matching on IDs
```

Comment:
Can you explain this in more detail? As a user I would be surprised that `compare_attrs=True` actually silently doesn't compare edge attributes based on another function argument. If doing it this way is the best option, we should document it well.

Reply:
It was unclear to me how to match edge attributes with each other if we are explicitly stating that we are ignoring the edge IDs in the first place. If you have a suggestion, happy to hear it, but my interim solution will be to (1) rename `compare_ids` to `compare_edge_ids` and (2) add a few sentences to the doc string.

Reply:
Totally! Is it possible or would it make sense to use the defaultdict/frozenset strategy again for attributes, or not really? That's the only possibility that comes to mind.

Reply:
hmmm won't every entry in this `defaultdict` just have a list of length 1? What is the data type of `H1._edge_attr[e]` and can we just use it as the value type for the defaultdict?

Also, what exactly will need to get sorted before comparison? Didn't quite understand that bit.

I agree it is a bit convoluted, but if we can do it correctly, then I think it makes sense to implement it rather than have a somewhat convoluted caveat in the docs!

Reply:
The issue would occur with multiedges. so the idea is that if I have three multi-edges `{1, 2, 3}` with IDs 0, 1, and 2 and associated attributes `color="red"`, `color="blue"`, and `color="yellow"`, then if I'm comparing to a hypergraph with the same edges but different edge IDs, how can I ensure that they are equal. And I guess that the way that makes sense to me to to construct a dict where keys are frozensets of edges and the values are lists of all associated attribute dictionaries. But the problem is that these lists may be out of order, so that's why I was thinking we might need to sort them. Of course this is not an issue if multi-edges aren't present, but since the `Hypergraph` object supports multi-edges, I figured that we should make sure this algo works with them.

Reply:
Thanks, that makes sense! I think it would be a good idea to implement this.

Reply:
> hmmm won't every entry in this `defaultdict` just have a list of length 1? What is the data type of `H1._edge_attr[e]` and can we just use it as the value type for the defaultdict?
> 
> Also, what exactly will need to get sorted before comparison? Didn't quite understand that bit.
> 
> I agree it is a bit convoluted, but if we can do it correctly, then I think it makes sense to implement it rather than have a somewhat convoluted caveat in the docs!

okay, I have implemented a working version of what you suggested. It's a little gross, but it works!

Reply:
I am thinking about whether we can we use the invariant established by the `edges*_with_counts` comparison to simplify into just one loop in the attribute comparison. My idea is that we already checked that the two hypergraphs have the same set of hyperedges appearing the same number of times, so maybe we can just use one loop over the hyperedges for this part and still achieve the same goal? Let me know if that makes any sense to you haha

Reply:
Do we maintain a hyperedge members -> hyperedge ids data structure? Can't remember off the top of my head, if we don't then the above idea won't be convenient since we need the IDs to get the attributes, so we can just forget it.

Reply:
I don't think that we do. If it's okay with you, I'll keep the two loops separate, because I don't know that it will save all that much time.